batch/batch/driver/instance_collection/job_private.py

```diff
@@ -235,58 +263,25 @@ def allocate_jobs(user, mark):
 
         return result
 
-    async def create_instance(self, batch_id, job_id, machine_spec):
-        assert machine_spec is not None
-
-        machine_name = self.generate_machine_name()
+    async def create_instance(self, machine_spec: dict) -> Tuple[Instance, List[QuantifiedResource]]:
         machine_type = machine_spec['machine_type']
         preemptible = machine_spec['preemptible']
         storage_gb = machine_spec['storage_gib']
-
-        machine_type_dict = machine_type_to_dict(machine_type)
-        cores = int(machine_type_dict['cores'])
-        cores_mcpu = cores * 1000
-        worker_type = machine_type_dict['machine_type']
-
-        zone = self.zone_monitor.get_zone(cores, False, storage_gb)
-        if zone is None:
-            return
-
-        activation_token = secrets.token_urlsafe(32)
-
-        config, worker_config = create_instance_config(
+        _, cores = self.resource_manager.worker_type_and_cores(machine_type)
+        instance, total_resources_on_instance = await self._create_instance(
             app=self.app,
-            zone=zone,
-            machine_name=machine_name,
+            cores=cores,
             machine_type=machine_type,
-            activation_token=activation_token,
-            max_idle_time_msecs=WORKER_MAX_IDLE_TIME_MSECS,
-            worker_local_ssd_data_disk=False,
-            worker_pd_ssd_data_disk_size_gb=storage_gb,
-            boot_disk_size_gb=self.boot_disk_size_gb,
-            preemptible=preemptible,
             job_private=True,
+            location=None,
+            preemptible=preemptible,
+            max_idle_time_msecs=None,
```

Comment:
Why did you make the max idle time here None?

Reply:
I suppose that's a bit confusing. None is the default value. I centralized that in InstanceCollection._create_instance. I'll inline the constant.

Reply:
Erm. I won't. I did it this way because in pool.py None is used to indicate the default max idea time. I pushed that convention everywhere for consistency